src/main/java/com/gotocompany/depot/maxcompute/converter/payload/TimestampPayloadConverter.java

```diff
@@ -0,0 +1,38 @@
+package com.gotocompany.depot.maxcompute.converter.payload;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Message;
+import com.gotocompany.depot.config.MaxComputeSinkConfig;
+import com.gotocompany.depot.maxcompute.converter.type.TimestampTypeInfoConverter;
+import lombok.RequiredArgsConstructor;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+
+@RequiredArgsConstructor
+public class TimestampPayloadConverter implements PayloadConverter {
+
+    private static final String SECONDS = "seconds";
+    private static final String NANOS = "nanos";
+
+    private final TimestampTypeInfoConverter timestampTypeInfoConverter;
+    private final MaxComputeSinkConfig maxComputeSinkConfig;
+
+    @Override
+    public Object convertSingular(Descriptors.FieldDescriptor fieldDescriptor, Object object) {
+        Message message = (Message) object;
+        long seconds = (long) message.getField(message.getDescriptorForType().findFieldByName(SECONDS));
+        int nanos = (int) message.getField(message.getDescriptorForType().findFieldByName(NANOS));
+        Instant instant = Instant.now();
+        ZoneOffset zoneOffset = maxComputeSinkConfig.getZoneId().getRules().getOffset(instant);
+        return LocalDateTime.ofEpochSecond(seconds, nanos, zoneOffset);
+    }
```

Comment:
Missing validation for invalid timestamp values (e.g., negative seconds/nanos). No handling for timestamps outside valid range. Should add proper bounds checking and error handling for invalid timestamps.

Reply:
Defer to phase 2, have discussion with higher ups

Reply:
I've added valid timestamp range and streaming event time checking